tsl/src/chunk.c

```diff
@@ -259,3 +262,60 @@ chunk_invoke_drop_chunks(Oid relid, Datum older_than, Datum older_than_type)
 
 	return num_results;
 }
+
+static bool
+chunk_is_distributed(const Chunk *chunk)
+{
+	return chunk->relkind == RELKIND_FOREIGN_TABLE;
+}
+
+Datum
+chunk_create_replica_table(PG_FUNCTION_ARGS)
+{
+	Oid chunk_relid;
+	const char *data_node_name;
+	const Chunk *chunk;
+	const Hypertable *ht;
+	const ChunkDataNode *cdn;
+	Cache *hcache = ts_hypertable_cache_pin();
+
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+
+	GETARG_NOTNULL_OID(chunk_relid, 0, "chunk");
+	GETARG_NOTNULL_NULLABLE(data_node_name, 1, "data node name", CSTRING);
+
+	chunk = ts_chunk_get_by_relid(chunk_relid, false);
+	if (chunk == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("relation \"%s\" is not a chunk", get_rel_name(chunk_relid))));
+	if (!chunk_is_distributed(chunk))
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("chunk \"%s\" doesn't belong to a distributed hypertable",
+						get_rel_name(chunk_relid))));
+
+	ht = ts_hypertable_cache_get_entry(hcache, chunk->hypertable_relid, CACHE_FLAG_NONE);
+	ts_hypertable_permissions_check(ht->main_table_relid, GetUserId());
+
+	/* Check the given data node exists */
+	data_node_get_foreign_server(data_node_name, ACL_USAGE, true, false);
```

Comment:
I don't recall what the `true` and `false` arguments mean (which is the problem of such boolean parameters), but I assume one of them means "fail if not found", correct?

Reply:
`true` is for `fail_on_aclcheck` and `false` is for `missing_ok`, i.e., opposite to fail if not found.

Reply:
Does it resolve your question?

Reply:
Suggestion: add a comment to explain that this fails, similar to the comment for the lookup below.

Reply:
I haven't seen that we add comments for internal API function calls to explain its arguments. I checked that none of the calls to this function has such comment. So the suggestion is not consistent with current practice and might be good to discuss on the team level.

Reply:
Just a suggestion for clarity since I didn't understand myself. Never intended for this to be controversial or requiring team-level discussions.